dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/ServiceConfig.java

```diff
@@ -657,18 +676,8 @@ private Integer findConfigedPorts(ProtocolConfig protocolConfig, String name, Ma
                 }
             }
         }
-
-        // save bind port, used as url's key later
-        map.put(Constants.BIND_PORT_KEY, String.valueOf(portToBind));
-
-        // registry port, not used as bind port by default
-        String portToRegistryStr = getValueFromConfig(protocolConfig, Constants.DUBBO_PORT_TO_REGISTRY);
-        Integer portToRegistry = parsePort(portToRegistryStr);
-        if (portToRegistry == null) {
-            portToRegistry = portToBind;
-        }
-
-        return portToRegistry;
+        protocolConfig.setPort(portToBind);

```

Comment:
method name (getPortFromProtocol) indicate a get operation here, should we change the state of input parameter?

Reply:
The change of the state of the input parameter is used to cached the parsed port in protocol config bean, so that the subsequent accesses of port of a protocol can retrieve the port from the protocol config bean directly. Or the find port logic will be executed during the exporting process of every <dubbo:service>